File: 01/Day01.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Collections;
import java.util.PriorityQueue;
import java.util.Scanner;

public class Day01 {
    private static PriorityQueue<Integer> getData(String fileName) {
        try {
            PriorityQueue<Integer> heap = new PriorityQueue<Integer>(Collections.reverseOrder());

            File tmp = new File(fileName);
            Scanner rdr = new Scanner(tmp);
            int current = 0;
            while (rdr.hasNextLine()) {
                String data = rdr.nextLine();
                if (data.length() == 0) {
                    heap.add(current);
                    current = 0;
                } else {
                    current += Integer.parseInt(data);
                }
            }

            heap.add(current);

            rdr.close();

            return heap;
        } catch (FileNotFoundException e) {
            System.out.println("File not found");
        }

        return null;
    }

    private static Integer puzzle1(PriorityQueue<Integer> heap) {
        return heap.peek();
    }

    private static Integer puzzle2(PriorityQueue<Integer> heap) {
        return heap.poll() + heap.poll() + heap.poll();
    }

    public static void main(String[] args) {
        PriorityQueue<Integer> sampleData = getData("./day01sample.txt");
        PriorityQueue<Integer> data = getData("./day01.txt");

        System.out.println("puzzle1");
        System.out.println(puzzle1(sampleData));
        System.out.println(puzzle1(data));

        System.out.println("puzzle2");
        System.out.println(puzzle2(sampleData));
        System.out.println(puzzle2(data));
    }
}

File: 01/day01.py
from heapq import heappop, heappush, heapify


def getData(fileName):
    data = open(fileName).read().strip()

    heap = []
    heapify(heap)

    current = 0
    for line in data.splitlines():
        if len(line) == 0:
            heappush(heap, -1 * current)
            current = 0
        else:
            current += int(line)

    heappush(heap, -1 * current)
    return heap

def puzzle1(heap):
    return -1 * heap[0]

def puzzle2(heap):
    return -1 * (heappop(heap) + heappop(heap) + heappop(heap) )

sampleData = getData("./day01sample.txt")
data = getData("./day01.txt")

print(puzzle1(sampleData))
print(puzzle1(data))
print("\n")

print(puzzle2(sampleData))
print(puzzle2(data))
